src/Undefinable/unwrapOr.ts

```diff
@@ -9,7 +9,7 @@ import { ERR_MSG_DEF_MUST_NOT_BE_NO_VAL_FOR_UNDEFINABLE } from './ErrorMessage';
  *  * _def_ must not be `Undefinable<*>`.
  *  * If the result of _def_ is `undefined`, throw `TypeError`.
  */
-export function unwrapOrFromUndefinable<T>(v: Undefinable<T>, def: T): T {
+export function unwrapOrFromUndefinable<T>(v: Undefinable<T>, def: NotUndefined<T>): NotUndefined<T> {
```

Comment:
This would be the following error

```typescript
declare function foo<S>(): Undefinable<S>;
declare function hoge<S>(): S;

function bar<T>(): T {
  const s: Undefinable<T> = foo<T>();
  // hoge<T>(), Argument of type 'T' is not assignable to parameter of type 'NotUndefined<T>'.ts(2345)
  return unwrapOrFromUndefinable<T>(s, hoge<T>());
}
```